draft-ietf-jsonpath-base.md

```diff
@@ -1036,28 +1036,30 @@ The current item is selected if and only if the result is `true`.
 
 
 ~~~~ abnf
+boolean-expr     = logical-or-expr / logical-and-expr
+logical-or-expr  = logical-and-expr "||" boolean-expr ; disjunction
+                                                      ; binds less tightly than conjunction
+logical-and-expr = basic-filter /
+                   basic-filter "&&" logical-and-expr ; conjunction
+                                                      ; binds more tightly than disjunction
 
-
-boolean-expr = logical-expr
-logical-expr = ([neg-op] primary-expr) / logical-or-expr
-neg-op       = "!"                                  ; not operator
-primary-expr = "(" logical-or-expr ")"
-logical-or-expr = logical-and-expr *["||" logical-and-expr]
-logical-and-expr = comp-expr *["&&" comp-expr]
+basic-filter = comp-expr / paren-expr / ([neg-op] paren-expr)
```

Comment:
`basic-filter` breaks the naming rule (not communicated though) of having `xxx-expr` and `yyy-op` exclusively here.

`basic-expr` or something would fit better, I think.

Reply:
Thanks for explaining the naming rule, which I hadn't noticed. I'll do some renaming.

Reply:
I went for `basic-expr`.